src/Numerics/MinimizeOneDim.h

```diff
@@ -0,0 +1,137 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2018 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_MINIMIZE_ONED_H
+#define QMCPLUSPLUS_MINIMIZE_ONED_H
+
+#include <algorithm>
+#include <stdexcept>
```

Comment:
Is there a problem with stdexcept?
The code throws a runtime exception if the minimization fails, compared to the original code which uses APP_ABORT.

Reply:
No. I was only worried about our super entangled header file inclusion. If that can be placed in cpp file, just don't put it in the header.

Reply:
No APP_ABORT in unit tests. We know handling MPI is complicated.

Reply:
I think headers should include what they use.  Otherwise it gets trickier to use a header file - need to know which other header files are required for it to work.

Reply:
I don't see stdexcept directly use here, why is it added here?

Reply:
Line 87, std::runtime_error is defined in stdexcept (according to the documentation)

Reply:
I see. I missed it. Then it is fine.